### PR TITLE
Don't expose macOS and iOS specific display servers.

### LINKED ISSUE #76717: GDScript autocomplete suggests nonexistent `DisplayServerMacOS` type
### Godot version

4.0.2.stable

### System information

macOS 12.6.3

### Issue description

`DisplayServerMacOS` seems to be a suggested type, but when I command-click for documentation, it brings up a blank page. I also can't find a reference to this on the online stable docs. Was this a removed type that got overlooked?

I asked some collaborators on Windows if they see this as well, and it seem like this option isn't suggested for them.

image

### Steps to reproduce

1. Open up a GDScript file on a macOS build of the IDE
2. Start typing "Display"

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by HolonProduction**
Fixes #76717.

### CODE REVIEW

**Comment by AThousandShips** [comment]
Maybe add a comment like the one in `DisplayServerWindows`:
`//No need to register, it's platform-specific and nothing is added`

**Comment by HolonProduction** [comment]
It's a bit inconsistent with other display servers. Android for example does not contain such a comment. But adding one there as well won't hurt.

**Comment by AThousandShips** [comment]
Windows and Linux/BSD do, so probably best to make it universal to prevent future incidents like this

**Comment by HolonProduction** [comment]
I added the comment to the remaining display servers. Hopefully I didn't miss one.

File: platform/android/display_server_android.h [hunk 1]
```diff
@@ -39,6 +39,9 @@ class RenderingDeviceVulkan;
 #endif
 
 class DisplayServerAndroid : public DisplayServer {
+	//No need to register, it's platform-specific and nothing is added
+	//GDCLASS(DisplayServerAndroid, DisplayServer)
```
Comment (akien-mga) [inline platform/android/display_server_android.h:43]:
```suggestion
	// No need to register with GDCLASS, it's platform-specific and nothing is added.
```

Comment (akien-mga) [reply platform/android/display_server_android.h:43]: Same for the others.

Comment (akien-mga) [reply platform/android/display_server_android.h]: You missed some changes from my suggestion:
- Space after `//`
- Sentence ends with a period

Comment (HolonProduction) [reply platform/android/display_server_android.h]: Fixed 😅

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**